### DISCUSSION TITLE
Feature: Syncing Workspaces

### DISCUSSION THREAD

**Discussion by mr-cheffy** [zen-browser/desktop#854] [Ideas]
### What feature would you like to see?

This can be split into multiple issues depending on the difficulty of implementing them all.

- [ ] Syncing Workspaces between computers

- [ ] Sync Tabs in each workspace between computers

### What platform are you seeing the problem on?

Linux

### Additional context

I'd find it useful to sync the workspaces so i don't have to create them multiple times on different devices.
 
 ~ _Created automatically from issue: https://github.com/zen-browser/desktop/issues/366_

**Comment by braccae** [comment]
Could this be accomplished with a hidden bookmarks folder? Something like

 <root_dir> -> _WORKSPACES -> <workspace_folder> -> open tabs

This way, it'd be possible to sync workspaces with a tool like Floccus or firefox sync and keep the workspaces consistent across open windows. One of the really nice features of Arc is that the workspaces are in the browser itself and not the window.

**Reply by cyrus01337** [reply]
+1 to this, wondering if it's possible to abduct a plugin's functionality altogether and hide it or something to that effect.

Another idea could be to do this in a custom way, though I'm unsure of how Firefox operates under the hood. If the current tabs are able to be serialised, then storing all of that in a collection file in say `~/.zen`

**Comment by HarunKilic** [comment]
This would be really nice. Which I also find really nice about Arc.

**Comment by kristijanribaric** [comment]
Workspaces sync has been implemented in `1.0.1-a.8`

**Reply by nicklayb** [reply]
I have a stupid question, are pinned tabs supposed to sync too? Cause I don't see them being synced here and I'm not sure I see the point of workspace syncing if that doesn't include the pinned tabs, is it just the name and icon that synced?

**Reply by kristijanribaric** [reply]
Currently only workspaces (name, theme and icon) are synced, you can expect pinned tabs sync in the near future

**Reply by SheepDomination** [reply]
Has this been updated to support all tabs per workspace ?

**Reply by JiangL1011** [reply]
It still doesn't seem to support the pinned tabs sync feature.

**Reply by jackTabsCode** [reply]
I'm confused. My workspaces don't sync between computers. I'm not even talking about the tabs, just the literal workspaces.

**Reply by RyKaT07** [reply]
> I'm confused. My workspaces don't sync between computers. I'm not even talking about the tabs, just the literal workspaces.

My too, I had to manually create then on another device, and now when I set one on one device then i disappear from another

**Comment by jbmorgado** [comment]
It doesn't sync the respective tabs open on them though.

I just get a mostly (odly there was 1 tab there) empty workspaces synced across devices.

**Reply by HarunKilic** [reply]
So do I. Thought I was doing it wrong. Tried to disconnect and also remove account on both PCs then reconnect. Without any luck. It does sync the workspaces tho.

**Comment by Mpic** [comment]
As of 1.19.1b, workspaces do not seem to sync at all, even empty of tabs... 
EDIT: It seems to be actively in development, see https://github.com/zen-browser/desktop/pull/12531 and https://github.com/zen-browser/desktop/pull/12507
